### PR TITLE
pre loading within SL

### PR DESCRIPTION
**Pull request by fupelaqu**
## Summary
A clear and concise description of the overall goals of the pull request's commits.

**Related Issue: #IssueId**

**PR Type: Bug Fix | Feature | Documentation**

**Status: WIP/Ready to review**

**Breaking change? Yes/No**

### Solution
Describe your changes in detail

### How has this been tested?
Outline the steps taken to test the PR (unit tests? Integration tests? tested impacts on other project features? ...)

### Other changes
Describe any other additional work done:
bug fixes, tweaks, small refactors

### Deploy Notes
Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, new environment variables, ...

### Remaining Todos
This should be done before merging this PR:
* [ ] - Task1
* [ ] - Task2

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] My code follows the code style of this project.
- [ ] I have updated the Release notes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/scala/ai/starlake/job/Main.scala [hunk 1]
```diff
@@ -218,14 +219,29 @@ class Main extends StrictLogging {
           if (settings.appConfig.forceHalt) {
             Runtime.getRuntime.halt(1)
           }
```
Comment (hayssams) [inline src/main/scala/ai/starlake/job/Main.scala:221]: to include in preload job

File: src/main/scala/ai/starlake/job/ingest/PreLoadCmd.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    config.strategy match {
+      case Some(PreLoadStrategy.Imported) =>
+        schemaHandler.domains(List(config.domain), config.tables.toList).headOption match {
+          case Some(domain) =>
+            Success(
+              PreLoadJobResult(
+                config.domain,
+                workflow(schemaHandler)
+                  .listStageFiles(domain, dryMode = true)
```
Comment (hayssams) [inline src/main/scala/ai/starlake/job/ingest/PreLoadCmd.scala:72]: pass access token

File: src/main/scala/ai/starlake/workflow/IngestionWorkflow.scala [hunk 3]
```diff
@@ -377,11 +383,147 @@ class IngestionWorkflow(
...
+          val tableAck = table.metadata
+            .flatMap(_.ack)
+            .orElse(domain.metadata.flatMap(_.ack))
+            .orElse(settings.appConfig.ack)
+            .getOrElse("")
+
+          val tableFiles =
+            domainFolderFilesInfo.filter(file => table.pattern.matcher(file.path.getName).matches())
+
+          if (tableAck.nonEmpty) {
```
Comment (hayssams) [inline src/main/scala/ai/starlake/workflow/IngestionWorkflow.scala]: rename to anyLevelAck

File: src/main/scala/ai/starlake/workflow/IngestionWorkflow.scala [hunk 4]
```diff
@@ -377,11 +383,147 @@ class IngestionWorkflow(
...
+            // for a given table, the files included within each archive that respect the table pattern
+            // should be added to the archived table files if and only if the corresponding table archived ack file exists
+            val archivedTableFiles =
+              archivedFiles
```
Comment (hayssams) [inline src/main/scala/ai/starlake/workflow/IngestionWorkflow.scala]: rename archived and archivedTable to zipped...

**Review by hayssams — APPROVED** [review]
LGTM

File: src/main/scala/ai/starlake/job/Main.scala (hunk 1 above)
Comment (fupelaqu) [reply src/main/scala/ai/starlake/job/Main.scala:221]: as discussed, needed if cmd called from an api

File: src/main/scala/ai/starlake/job/ingest/PreLoadCmd.scala (hunk 2 above)
Comment (fupelaqu) [reply src/main/scala/ai/starlake/job/ingest/PreLoadCmd.scala:72]: don't think accessToken is required here, I should remove it from the preload config. @hayssams what do you think ?

Comment (hayssams) [reply src/main/scala/ai/starlake/job/ingest/PreLoadCmd.scala:72]: We need a way or another when this is run from the command line.
Or the credentials are the ones in the env or the token is passed as a parameter
If it is started from Airflow for example, how do we pass the identity to the preload executable

**Pull request merged**